lib/kubeclient/common.rb

```diff
@@ -296,11 +296,15 @@ def get_entity(klass, resource_name, name, namespace = nil)
       new_entity(result, klass)
     end
 
-    def delete_entity(resource_name, name, namespace = nil)
+    def delete_entity(resource_name, name, namespace = nil, options = nil)
       ns_prefix = build_namespace_prefix(namespace)
+
+      params = {}
+      params.merge!(options[:query]) if options.include?(:query)
```

Comment:
We do things a little differently in get_entitites(pass only the relevant options and not a nested hash), please take a look.

why do we currently only support a limited set of url parameters, and not pass in every thing that we get?
(get a 'params' hash and pass that in)
thoughts @simon3z @abonas @cben ?